api/src/main/java/org/apache/iceberg/geospatial/BoundingBox.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.geospatial;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Represents a geospatial bounding box composed of minimum and maximum bounds.
+ *
+ * <p>A bounding box (also called a Minimum Bounding Rectangle or MBR) is defined by two points: the
+ * minimum and maximum coordinates that define the box's corners. This provides a simple
+ * approximation of a more complex geometry for efficient filtering and data skipping.
+ */
+public class BoundingBox {
+  /**
+   * Create a {@link BoundingBox} object from buffers containing min and max bounds
+   *
+   * @param min the serialized minimum bound
+   * @param max the serialized maximum bound
+   * @return a BoundingBox instance
+   */
+  public static BoundingBox fromByteBuffers(ByteBuffer min, ByteBuffer max) {
+    return new BoundingBox(
+        GeospatialBound.fromByteBuffer(min), GeospatialBound.fromByteBuffer(max));
+  }
+
+  /**
+   * Deserialize a byte buffer as a {@link BoundingBox} object
+   *
+   * @param buffer the serialized bounding box
+   * @return a BoundingBox instance
+   */
+  public static BoundingBox fromByteBuffer(ByteBuffer buffer) {
+    ByteBuffer tmp = buffer.duplicate();
```

Comment:
I know @huaxingao suggested the `duplicate` to avoid mutating the input arg, which I agree. 

But I am wondering if we just need to add a `Preconditions.checkArgument` assertion here on the byte order?

Reply:
@Kontinuation this is my last comment. will approve after this is addressed.

Reply:
I know it's a shallow copy. It is an object allocation. Byte buffer object itself is probably larger than the column stats bytes (4, 8, 16 bytes etc.). For wide table with a lot of column stats, this is not trivial overhead. But I agree that we we should keep `ByteBuffer.duplicate()` following the `Conversions` style to avoid pre-mature optimization. 

Avoiding mutating the caller's buffer is a good argument for keeping `duplicate`.